packages/@aws-cdk/aws-ec2-alpha/lib/vpc-v2-base.ts

```diff
@@ -196,6 +323,106 @@ export abstract class VpcV2Base extends Resource implements IVpcV2 {
     });
   }
 
+  /**
+   * Adds a new Egress Only Internet Gateway to this VPC and defines a new route
+   * to the route table of given subnets
+   * @default in case of no input subnets, no route is created.
+   */
+  public addEgressOnlyInternetGateway(options?: EgressOnlyInternetGatewayOptions): void {
+    const egw = new EgressOnlyInternetGateway(this, 'EgressOnlyGW', {
+      vpc: this,
+    });
+
+    const useIpv6 = (this.secondaryCidrBlock.some((secondaryAddress) => secondaryAddress.amazonProvidedIpv6CidrBlock === true ||
+    secondaryAddress.ipv6IpamPoolId != undefined))? true : false;
+
+    if (!useIpv6) {
+      throw new Error('Egress only IGW can only be added to Ipv6 enabled VPC');
+    };
+
+    if (options?.subnets) {
+      const subnets = flatten(options.subnets.map(s => this.selectSubnets(s).subnets));
+      subnets.forEach((subnet) => {
+        this.createEgressRoute(subnet, egw, options.destination);
```

Comment:
What is this doing?

Reply:
this is the default route that we are creating under provided subnet's routeTable. It will add an entry to subnet's routeTable either with the specified destination address or with default(::/0)